New Resource: `azurerm_kusto_database_principal_assignment`

jrauschenbusch opened the pull request:
fixes https://github.com/terraform-providers/terraform-provider-azurerm/issues/7383

jrauschenbusch commented:
@katbyte This is currently WIP. I'll need to check whether everything works correctly. After all we need to deprecate the old "azurerm_database_principal" resource and give a hint to the new available resource within the docs.

jrauschenbusch commented on azurerm/internal/services/kusto/kusto_database_principal_assignment_resource.go:
```diff
@@ -0,0 +1,248 @@
+	return nil
+}
+
+func validateAzureRMKustoDatabasePrincipalAssignmentName(v interface{}, k string) (warnings []string, errors []error) {
```
Naming rules must be verified. Currently the same as Database naming.

jrauschenbusch commented:
@mbfrahry As it's the same structure as `azurerm_kusto_cluster_principal_assignment` it could make sense that you also review this one. Notes from the other PR are already applied.

jackofallops commented on website/docs/r/kusto_database_principal_assignment.html.markdown:
```diff
@@ -0,0 +1,100 @@
+
+* `create` - (Defaults to 1 hour) Used when creating the Kusto Database Principal Assignment.
+* `read` - (Defaults to 5 minutes) Used when retrieving the Kusto Database Principal Assignment.
+* `update` - (Defaults to 1 hour) Used when updating the Kusto Database Principal Assignment.
```
can remove
```suggestion
```

jrauschenbusch commented:
@jackofallops I've fixed everything and added the `requiresImport` acc test. And yes, everything was based on the `azurerm_kusto_cluster_principal_assignment` resource.

Please review again. I had to do a `go mod vendor` which updated the vendors folder. I don't know if the diffs should really be there. Please consider it within the review.

jrauschenbusch commented on vendor/github.com/hashicorp/go-azure-helpers/authentication/config.go:
```diff
@@ -14,6 +14,7 @@ import (
 // new Azure management client.
 type Config struct {
 	ClientID           string
+	ClientSecret       string
```
This diff is definitively not intended to be here.

jackofallops replied:
Thanks @jrauschenbusch - I'll fetch your branch and figure out what's going on, we had to make an emergency fix to the master branch on Monday which is likely to be the root of the issue.

jackofallops replied:
I _think_ it's sorted now, will let Travis do it's thing and loop back shortly, we're a bit stretched today.

jackofallops commented:
Tests passing:
![image](https://user-images.githubusercontent.com/11830746/86941810-4949e700-c13c-11ea-8b67-6b1bcd1e7cd2.png)
